api/types/authentication.go

```diff
@@ -701,3 +714,88 @@ func (d *MFADevice) MarshalJSON() ([]byte, error) {
 func (d *MFADevice) UnmarshalJSON(buf []byte) error {
 	return jsonpb.Unmarshal(bytes.NewReader(buf), d)
 }
+
+// RequireMFAType is the type of MFA requirement, such as per-session MFA
+// or per-request PIV touch.
+type RequireMFAType string
+
+const (
+	// RequireMFAOff means MFA is *not* required to begin server sessions.
+	RequireMFAOff = RequireMFAType("off")
+	// RequireSessionMFA means MFA is required to begin server sessions.
+	RequireSessionMFA = RequireMFAType("session_mfa")
+	// RequireSessionMFAAndHardwareKey means MFA is required to begin server sessions,
+	// and login sessions must use a private key backed by a hardware key.
+	RequireSessionMFAAndHardwareKey = RequireMFAType("session_mfa_and_hardware_key")
+	// RequireHardwareKeyTouch means login sessions must use a hardware private key that
+	// requires touch to be used. This touch requirement applies to all API requests
+	// rather than only session requests. This touch is different from MFA, so to prevent
+	// requiring double touch on session requests, normal Session MFA is disabled.
+	RequireHardwareKeyTouch = RequireMFAType("hardware_key_touch")
+)
+
+// IsSessionMFARequired returns whether this RequireMFAType requires per-session MFA.
+func (r RequireMFAType) IsSessionMFARequired() bool {
+	return r == RequireSessionMFA || r == RequireSessionMFAAndHardwareKey
+}
+
+// UnmarshalYAML supports parsing RequireMFAType from boolean or alias.
+func (r *RequireMFAType) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var val interface{}
+	err := unmarshal(&val)
+	if err != nil {
+		return err
+	}
+
+	if *r, err = parseRequireMFAType(val); err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+// UnmarshalJSON supports parsing RequireMFAType from boolean or alias.
+func (r *RequireMFAType) UnmarshalJSON(data []byte) error {
+	var val interface{}
+	err := json.Unmarshal(data, &val)
+	if err != nil {
+		return err
+	}
+
+	if *r, err = parseRequireMFAType(val); err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+func parseRequireMFAType(val interface{}) (RequireMFAType, error) {
+	switch v := val.(type) {
+	case string:
+		switch v {
+		case string(RequireMFAOff), "":
+			return RequireMFAOff, nil
+		case string(RequireSessionMFA):
+			return RequireSessionMFA, nil
+		case string(RequireSessionMFAAndHardwareKey), "hardware_key":
```

Comment:
I understood from your PR that "hardware_key" is an alias, but could we make this a `RequireMFAType` constant like the others?

Reply:
Thanks for pointing this out, I think this alias idea doesn't make sense after all. 

My intention was to make the functionality more clear internally, but that can be handled just by making the variable `RequireMFASessionAndHardwareKey = RequireMFAType("hardware_key")`.